File: frontend/lib/types/userData.ts
import { DocumentReference } from "firebase/firestore"

/**
 * Data schema for User Data, separate from the user data directly from our authentication server.
 */
export default interface UserData {
    display_name: string,
    email: string,
    registered_date: string,
    admin: string,
    posts: DocumentReference[],
    donations_made: Number
}

File: backend/types/user-data.go
package types

import (
	"time"

	"cloud.google.com/go/firestore"
)

// UserData represents a user's data.
// It includes display name, email, registration timestamp, admin status, user's posts,
// UID and count of donations made.
type UserData struct {
	DisplayName           string                   `json:"display_name"`
	Email                 string                   `json:"email"`
	RegistrationTimestamp time.Time                `json:"registered_date"` // In UTC
	Admin                 bool                     `json:"admin"`
	Posts                 []*firestore.DocumentRef `json:"posts"`
	UID                   string                   `json:"uid"`
	DonationsMade         int64                    `json:"donations_made"`
}
